docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc

```diff
@@ -440,3 +452,34 @@ your entity `Person` by typing `Person.` and getting completion for all the oper
 `Person.find("name", "stef")`.
 
 That's all there is to it: with Panache, Hibernate ORM has never looked so trim and neat. 
+
+== Defining entities in external projects or jars
+
+Hibernate ORM with Panache in Quarkus relies on compile-time bytecode enhancements to your entities. If you define your entities in the
+same project where you build your Quarkus application, everything will work fine. If the entities come from external projects
+or jars, you should build those jars with the Jandex Maven plugin, so that your entities will be indexed and found by Quarkus
```

Comment:
Adding beans.xml to the archive will also do this.

Reply:
Does it really? I thought it was only about CDI beans and not about Hibernate entities.

Reply:
I just tried it and it does not work with just a `beans.xml` file.

Reply:
We should be indexing anything with a META-INF/beans.xml file. Anything with this file is considered to be an 'application archive' (same as if it had a jandex index). 

beans.xml does have additional meaning from a CDI point of view, but if there are no classes with bean defining annotations then the result should be the same.

Reply:
Well, Ok, but this is not the case currently, and this PR is about docs. Pity that it missed the release, too :(

Reply:
https://github.com/quarkusio/quarkus/issues/2705